### PR TITLE
UPDATE: Dynamic Card height

### PR DESCRIPTION
**Pull request by tomashapl**
This Pull Request meets the following criteria:

* [ ] Tests have been added/adjusted for my new feature
* [ ] New Components are registered in index.js of my project
<br/><br/><br/><url>LiveURL: https://orbit-components-card-dynamic-height.surge.sh</url>

### CODE REVIEW

File: src/utils/Slide/index.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  getSnapshotBeforeUpdate(prevProps) {
+    if (this.props.expanded === prevProps.expanded) return null;
+    return true;
+  }
+
+  componentDidUpdate(prevProps, prevState, snapshot) {
+    if (snapshot) {
+      if (this.props.expanded) {
+        this.setMaxHeight();
+        setTimeout(this.expandCallback, 150);
```
Comment (wEprous) [inline src/utils/Slide/index.js]: Don't forget to clear the timeouts 🙂

File: src/Card/CardSection/CardSectionContent/index.js [hunk 2]
```diff
@@ -33,11 +21,13 @@ export const StyledCardSectionContent = styled.div`
...
   padding-top: ${({ theme, expandable, expanded, visible }) =>
     hasPaddingTop({ expandable, expanded, visible }) && theme.orbit.spaceLarge};
-  transition: max-height ${({ theme }) => theme.orbit.durationFast} linear,
-    padding ${({ theme }) => theme.orbit.durationFast} linear,
+  transition: padding ${({ theme }) => theme.orbit.durationFast} linear,
     border-top ${({ theme }) => theme.orbit.durationFast} linear;
-  max-height: ${getMaxHeight};
   overflow: hidden;
+
+  ${StyledSlide} {
```
Comment (wEprous) [inline src/Card/CardSection/CardSectionContent/index.js]: Really necessary, can be this done by `Slide` component itself? 🤔

Comment (tomashapl) [reply src/Card/CardSection/CardSectionContent/index.js]: Yes, it is case if CardContent is default visible.

This is little bit different behaviour then obviously.

image

Comment (wEprous) [reply src/Card/CardSection/CardSectionContent/index.js]: Kay

**Pull request merged**